internal/thelma/state/providers/sherlock/state_loader.go

```diff
@@ -0,0 +1,145 @@
+package sherlock
+
+import (
+	"github.com/broadinstitute/thelma/internal/thelma/clients/sherlock"
+	"github.com/broadinstitute/thelma/internal/thelma/state/api/terra"
+	"github.com/broadinstitute/thelma/internal/thelma/utils/shell"
+)
+
+type stateLoader struct {
+	sherlock    sherlock.StateLoader
+	shellRunner shell.Runner
+	thelmaHome  string
+}
+
+func NewStateLoader(thelmaHome string, shellRunner shell.Runner, sherlock sherlock.StateLoader) terra.StateLoader {
+	return &stateLoader{
+		thelmaHome:  thelmaHome,
+		shellRunner: shellRunner,
+		sherlock:    sherlock,
+	}
+}
+
+func (s *stateLoader) Load() (terra.State, error) {
+	stateClusters, err := s.sherlock.Clusters()
+	if err != nil {
+		return nil, err
+	}
+
+	stateEnvironments, err := s.sherlock.Environments()
+	if err != nil {
+		return nil, err
+	}
+
+	// transforms cluster data returned by sherlock client into thelma's state domain types
+	clusters, err := s.buildClustersState(stateClusters)
+	if err != nil {
+		return nil, err
+	}
+
+	// transforms environment data returned by sherlock client into thelma's state domain types
+	environments, err := s.buildEnvironmentsState(stateEnvironments, clusters)
+	if err != nil {
+		return nil, err
+	}
+
+	return &state{
+		sherlock:     s.sherlock,
+		environments: environments,
+		clusters:     clusters,
+	}, nil
+}
+
+func (s *stateLoader) buildClustersState(clusters sherlock.Clusters) (map[string]*cluster, error) {
+	result := make(map[string]*cluster)
+	for _, cl := range clusters {
+		releases := make(map[string]*clusterRelease)
+		c := &cluster{
+			address:       cl.Address,
+			googleProject: cl.GoogleProject,
+			destination: destination{
+				name:            cl.Name,
+				base:            cl.Base,
+				requireSuitable: *cl.RequiresSuitability,
+				destinationType: terra.ClusterDestination,
+			},
+		}
+		stateReleases, err := s.sherlock.ClusterReleases(cl.Name)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, r := range stateReleases {
+			releases[r.Name] = &clusterRelease{
+				release: release{
+					name:         r.Name,
+					enabled:      true,
+					releaseType:  terra.ClusterReleaseType,
+					chartVersion: r.ChartVersionExact,
+					chartName:    r.Chart,
+					repo:         *r.ChartInfo.ChartRepo,
+					namespace:    r.Namespace,
+					helmfileRef:  r.HelmfileRefOrDefault("HEAD"),
```

Comment:
Same note as above, the default here won't ever get used (Sherlock goes so far as to prevent even it being explicitly set to an empty string)